### DISCUSSION TITLE
Implementing custom listener and datadog integration

### DISCUSSION THREAD

**Discussion by madhusudhan-patha** [serenity-bdd/serenity-core#2744] [Q&A]
We are having around 5 thousand tests and currently we need to wait till the execution is finished to get the reports. We are planning for live reporting (live results) with help of datadog to see the results as and when a test case fails . One way we are planning is to implement custom listener. Is it possible to implement a custom listener and register with CucumberWithSerenity?

**Comment by wakaleo** [comment]
Yes, you could do that.

**Reply by madhusudhan-patha** [reply]
@wakaleo can you please provide any links or references to it?

**Comment by wakaleo** [comment]
Have a look at the ConsoleStepListener class. You can register a listener using `StepEventBus.getEventBus().registerListener()`.

**Comment by madhusudhan-patha** [comment]
Thank you @wakaleo I followed above instructions. I am able to listen to only stepStarted, stepFinished events, but not able to listen on testStarted, testFinished, scenarioStarted etc. Any suggestions please?

We are trying to run the tests with CucumberWithSerenity runner + Junit. And registered the listener inside BeforeScenario annotation.
